tests/SpeechRecognizerTests.ts

```diff
@@ -355,6 +401,84 @@ describe.each([true, false])("Service based tests", (forceNodeWebSocket: boolean
         });
     });
 
+    test("testGetAutoDetectSourceLanguage", (done: jest.DoneCallback) => {
+        // tslint:disable-next-line:no-console
+        console.info("Name: testGetAutoDetectSourceLanguage");
+
+        const s: sdk.SpeechConfig = BuildSpeechConfig();
+        objsToClose.push(s);
+
+        const r: sdk.SpeechRecognizer = BuildRecognizerFromAutoConfig(s);
+        objsToClose.push(r);
+
+        r.canceled = (o: sdk.Recognizer, e: sdk.SpeechRecognitionCanceledEventArgs): void => {
+            try {
+                expect(e.errorDetails).toBeUndefined();
+            } catch (error) {
+                done.fail(error);
+            }
+        };
+
+        r.recognizeOnceAsync((result: sdk.SpeechRecognitionResult) => {
+            try {
+                expect(result).not.toBeUndefined();
+                expect(result.errorDetails).toBeUndefined();
+                expect(result.text).toEqual(Settings.WaveFileText);
+                expect(result.properties).not.toBeUndefined();
+                expect(result.properties.getProperty(sdk.PropertyId.SpeechServiceResponse_JsonResult)).not.toBeUndefined();
+                expect(result.language).not.toBeUndefined();
+                expect(result.languageDetectionConfidence).not.toBeUndefined();
+
+                done();
+            } catch (error) {
+                done.fail(error);
+            }
+        }, (error: string) => {
+            done.fail(error);
+        });
+    });
+    test("testRecognizeFromSourceLanguageConfig", (done: jest.DoneCallback) => {
+        // tslint:disable-next-line:no-console
+        console.info("Name: testRecognizeFromSourceLanguageConfig");
+
+        const s: sdk.SpeechConfig = BuildSpeechConfig();
+        objsToClose.push(s);
+
+        const configs: sdk.SourceLanguageConfig[] = BuildSourceLanguageConfigs();
+        configs.forEach((c: sdk.SourceLanguageConfig) => { objsToClose.push(c); });
+
+        const a: sdk.AutoDetectSourceLanguageConfig = BuildAutoConfig(configs);
+        objsToClose.push(a);
+
+        const r: sdk.SpeechRecognizer = BuildRecognizerFromAutoConfig(s, a);
```

Comment:
>BuildRecognizerFromAutoConfig [](start = 40, length = 29)

General test comments:
Can you fork these into a different file to help w/ parallelism?
Add a test for continuous reco.
Because the speech.context has to be retransmitted correctly each turn, add a test that spans turns. (Either by start / stop cont or inject enough silence the service ends the turn) The later may be easier as there's existing examples of how to use a PullStream to achieve that effect.

Reply:
Separated and added a test.